### ISSUE TITLE
``test_bdb`` leaks references

### ISSUE THREAD

**Issue by Eclips4** [python/cpython#118605]
# Bug report

### Bug description:

```python
 ./python -m test -R 3:3 test_bdb
Using random seed: 2592309756
0:00:00 load avg: 25.13 Run 1 test sequentially
0:00:00 load avg: 25.13 [1/1] test_bdb
beginning 6 repetitions. Showing number of leaks (. for 0 or less, X for 10 or more)
123:456
XX2 222
test_bdb leaked [1, 1, 1] references, sum=3
test_bdb leaked [2, 2, 2] memory blocks, sum=6
test_bdb failed (reference leak)

== Tests result: FAILURE ==

1 test failed:
    test_bdb

Total duration: 845 ms
Total tests: run=35
Total test files: run=1/1 failed=1
Result: FAILURE
```

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

Linux

### Linked PRs
* gh-118607

**Comment by gaogaotiantian** [comment]
If this is new, it's possible that it was caused by #111744. I'll take a look at it.

**Comment by gaogaotiantian** [comment]
Fixed in #118607

**Comment by Eclips4** [comment]
Thank you Gao for fixing this!
